Unable to install on a Cray (Uppercase module files).

When trying to install on a Cray, it fails. As by default the cray compiler `ftn` writes the module files it creates as uppercase (e.g. `PIO.mod`). 

```
crystal:/lus/scratch/n17843/ParallelIO-pio2_3_0/build> make install
[ 40%] Built target pioc
[ 54%] Built target genf90
[ 86%] Built target piof
[ 90%] Built target examplePio_f90
[ 95%] Built target example1
[100%] Built target darray_no_async
[100%] Built target darray_async
Install the project...
-- Install configuration: ""
-- Installing: /lus/scratch/n17843/sw/pio2/lib/libpioc.a
-- Up-to-date: /lus/scratch/n17843/sw/pio2/include/pio.h
-- Installing: /lus/scratch/n17843/sw/pio2/lib/libpiof.a
CMake Error at src/flib/cmake_install.cmake:40 (file):
  file INSTALL cannot find
  "/lus/scratch/n17843/ParallelIO-pio2_3_0/build/src/flib/pio.mod".
Call Stack (most recent call first):
  src/cmake_install.cmake:38 (include)
  cmake_install.cmake:37 (include)
  


Makefile:93: recipe for target 'install' failed
make: *** [install] Error 1
```

When looking at the module files:
```
crystal:/lus/scratch/n17843/ParallelIO-pio2_3_0/build> ls src/flib/
CMakeFiles           libpiof.a      PIO_KINDS.mod     PIONFATT_MOD.mod  pionfput_mod.F90  tryFortran_CSIZEOF
cmake_install.cmake  Makefile       PIOLIB_MOD.mod    pionfget_mod.F90  PIONFPUT_MOD.mod  tryMPI_HAS_Fortran_MOD
dtypes.h             piodarray.F90  PIO.mod           PIONFGET_MOD.mod  PIO_SUPPORT.mod
genf90               PIODARRAY.mod  pionfatt_mod.F90  PIO_NF.mod        PIO_TYPES.mod
```

--- a/src/flib/CMakeLists.txt
+++ b/src/flib/CMakeLists.txt
@@ -55,6 +55,8 @@ elseif (CMAKE_Fortran_COMPILER_ID STREQUAL "NAG")
   set ( CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -mismatch_all" )
   #    target_compile_options (piof
   #        PRIVATE -mismatch_all)
+elseif (CMAKE_Fortran_COMPILER_ID STREQUAL "Cray")
+  set (CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -ef")
 endif()
 
 # Look for c_sizeof capability

